From d991b5649f9224033494a8a7908aeccc966e2fc9 Mon Sep 17 00:00:00 2001
From: Ales Novak <alnovak@suse.cz>
Date: Thu, 25 Jun 2015 11:53:13 +0200
Subject: [PATCH] fix: lpfc_send_rscn_event sends bigger buffer size
Patch-mainline: submitted
References: bnc#935757

lpfc_send_rscn_event allocates data for "lpfc_rscn_event_header +
payload", but claims that the data has size of "lpfc_els_event_header +
payload". That leads to buffer overruns.

Signed-off-by: Ales Novak <alnovak@suse.cz>
---
 drivers/scsi/lpfc/lpfc_els.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/lpfc/lpfc_els.c b/drivers/scsi/lpfc/lpfc_els.c
index 0c4aea5..2ee1d19 100644
--- a/drivers/scsi/lpfc/lpfc_els.c
+++ b/drivers/scsi/lpfc/lpfc_els.c
@@ -4747,7 +4747,7 @@ lpfc_send_rscn_event(struct lpfc_vport *vport,
 
 	fc_host_post_vendor_event(shost,
 		fc_get_event_number(),
-		sizeof(struct lpfc_els_event_header) + payload_len,
+		sizeof(struct lpfc_rscn_event_header) + payload_len,
 		(char *)rscn_event_data,
 		LPFC_NL_VENDOR_ID);
 
-- 
2.4.1.168.g1ea28e1

